controller/src/main/java/io/pravega/controller/server/eventProcessor/requesthandlers/CommitTransactionTask.java

```diff
@@ -0,0 +1,175 @@
+/**
+ * Copyright (c) 2017 Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.controller.server.eventProcessor.requesthandlers;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Preconditions;
+import io.pravega.common.concurrent.Futures;
+import io.pravega.controller.store.stream.OperationContext;
+import io.pravega.controller.store.stream.StoreException;
+import io.pravega.controller.store.stream.StreamMetadataStore;
+import io.pravega.controller.store.stream.TxnStatus;
+import io.pravega.controller.task.Stream.StreamMetadataTasks;
+import io.pravega.shared.controller.event.CommitEvent;
+import lombok.extern.slf4j.Slf4j;
+
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.stream.Collectors;
+
+/**
+ * Request handler for processing commit events in request-stream.
+ */
+@Slf4j
+public class CommitTransactionTask implements StreamTask<CommitEvent> {
+    private final StreamMetadataTasks streamMetadataTasks;
+    private final StreamMetadataStore streamMetadataStore;
+    private final ScheduledExecutorService executor;
+    private final BlockingQueue<CommitEvent> processedEvents;
+
+    public CommitTransactionTask(final StreamMetadataStore streamMetadataStore,
+                                 final StreamMetadataTasks streamMetadataTasks,
+                                 final ScheduledExecutorService executor) {
+        this(streamMetadataStore, streamMetadataTasks, executor, null);
+    }
+
+    @VisibleForTesting
+    public CommitTransactionTask(final StreamMetadataStore streamMetadataStore,
+                                 final StreamMetadataTasks streamMetadataTasks,
+                                 final ScheduledExecutorService executor,
+                                 final BlockingQueue<CommitEvent> queue) {
+        Preconditions.checkNotNull(streamMetadataStore);
+        Preconditions.checkNotNull(streamMetadataTasks);
+        Preconditions.checkNotNull(executor);
+        this.streamMetadataStore = streamMetadataStore;
+        this.streamMetadataTasks = streamMetadataTasks;
+        this.executor = executor;
+        this.processedEvents = queue;
+    }
+
+    @Override
+    public CompletableFuture<Void> execute(CommitEvent event) {
+        // 1. get epoch.. check if the epoch matches the epoch in commit event.. if true, we can proceed.. else throw OperationNotAllowedException
```

Comment:
As a style point:
Comments can be out of date or wrong bug code cannot.
Instead of putting a comment like this indicating what is going to happen in order, it is better to create 6 tiny private methods for each of these steps and have this method call each in turn. This will add the same clarity, but it can never be out of date or wrong.

Reply:
You are right about the style suggestion. The code is structured to have multiple private methods each doing one small thing. This comment here is descriptive of what the code is doing at an uber level. I will move this inside those code blocks so that they continue to stay relevant wrt changing code.